Jint/Native/Array/ArrayPrototype.cs

```diff
@@ -803,71 +804,14 @@ private JsValue Sort(JsValue thisObj, JsValue[] arguments)
                 return obj.Target;
             }
 
-            int Comparer(JsValue x, JsValue y)
-            {
-                if (ReferenceEquals(x, null))
-                {
-                    return 1;
-                }
-
-                if (ReferenceEquals(y, null))
-                {
-                    return -1;
-                }
-
-                var xUndefined = x.IsUndefined();
-                var yUndefined = y.IsUndefined();
-                if (xUndefined && yUndefined)
-                {
-                    return 0;
-                }
-
-                if (xUndefined)
-                {
-                    return 1;
-                }
-
-                if (yUndefined)
-                {
-                    return -1;
-                }
-
-                if (compareFn != null)
-                {
-                    var s = TypeConverter.ToNumber(compareFn.Call(Undefined, new[] {x, y}));
-                    if (s < 0)
-                    {
-                        return -1;
-                    }
-
-                    if (s > 0)
-                    {
-                        return 1;
-                    }
-
-                    return 0;
-                }
-
-                var xString = TypeConverter.ToString(x);
-                var yString = TypeConverter.ToString(y);
-
-                var r = CompareOrdinal(xString, yString);
-                return r;
-            }
+            
 
-            var array = new JsValue[len];
-            for (uint i = 0; i < (uint) array.Length; ++i)
-            {
-                var value = obj.TryGetValue(i, out var temp)
-                    ? temp
-                    : null;
-                array[i] = value;
-            }
+            JsValue[] array;
 
             // don't eat inner exceptions
             try
             {
-                System.Array.Sort(array, Comparer);
+                array = obj.OrderBy(x => x, new ArrayComparer(compareFn)).ToArray();
```

Comment:
This will have performance implications 😉

Reply:
Before it was creating a copy of the array already, if this is what you mean.
We can't alter the compare function result and use the Array.Sort since it can invoke it like (a,b) or (b,a).
Any other option in mind?

Reply:
Well LINQ probably is a bit slower, but probably some benchmark would show if there's something to worry about. Unfortunately we don't seem to have a test case for `Array.sort` 🙁 So this is probably fine...